### PR TITLE
Java8 stuff

### PR DESCRIPTION
**Pull request by joaoandremartins**
Java 8 changes previously discussed in #1258, if we want to submit them.

### CODE REVIEW

**Comment by joaoandremartins** [comment]
This PR is supposed to be merged to clear-build-log though, for some reason, Github's diff is clearly master..

File: google-account-plugin/testSrc/com/google/cloud/tools/intellij/stats/UsageTrackerManagerTest.java [hunk 1]
```diff
@@ -34,7 +36,7 @@
  * Tests for {@link UsageTrackerManager}.
  */
 @RunWith(MockitoJUnitRunner.class)
-public class UsageTrackerManagerTest extends TestCase {
+public class UsageTrackerManagerTest {
```
Comment (patflynn) [inline google-account-plugin/testSrc/com/google/cloud/tools/intellij/stats/UsageTrackerManagerTest.java]: @joaoandremartins could you make this change in a separate PR? I don't want to wait to merge this and get rid of the warning.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java [hunk 2]
```diff
@@ -105,7 +106,8 @@ public void apply() {
 
       if (!libsToAdd.isEmpty()) {
         for (AppEngineStandardMavenLibrary library : libsToAdd) {
-          AppEngineSupportProvider.loadMavenLibrary(context.getModule(), library);
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: I think this was already updated in master. Could you do the rebase fix?

File: google-account-plugin/testSrc/com/google/cloud/tools/intellij/stats/UsageTrackerManagerTest.java (hunk 1 above)
Comment (joaoandremartins) [inline google-account-plugin/testSrc/com/google/cloud/tools/intellij/stats/UsageTrackerManagerTest.java]: :+1:

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java (hunk 2 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: Wish me good luck!!

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java [hunk 3]
```diff
@@ -167,50 +168,42 @@ private boolean isManagedLibrariesEnabled() {
...
 
-      if (library == null) {
-        return;
-      }
+      libraryOptional.ifPresent(
+          library -> {
+            final DependencyScope scope = library.getScope();
 
-      final DependencyScope scope = library.getScope();
+            ApplicationManager.getApplication().runWriteAction(() -> {
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: Just to be safe, did you test this out in the UI to make sure runWriteAction does the same thing as before?

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: I created new projects with all Maven dependencies checked, and verified that all of the dependencies were there.

Since you're more familiar with this code, do you know of other useful tests I can run?

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: You should also test adding them from the facet editor after the project is already there. Adding one at a time, adding multiple at once, removing them, then re-adding them etc.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java [hunk 4]
```diff
@@ -167,50 +168,42 @@ private boolean isManagedLibrariesEnabled() {
...
 
     @Override
     public void afterLibraryRemoved(Library removedLibrary) {
-      AppEngineStandardMavenLibrary library = AppEngineStandardMavenLibrary
-          .getLibraryByMavenDisplayName(removedLibrary.getName());
-
-      if (library != null) {
-        appEngineStandardLibraryPanel.toggleLibrary(library, false /* select */);
-      }
+      AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(removedLibrary.getName())
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: minor, but for consistency with the other method could you extract a variable here too?

Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: I don't think it's really necessary here. It's pretty readable this way, no?

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: sure. I was just thinking for consistency, but i'm ok.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java (hunk 3 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: All those tests succeeded for me.

One thing I noticed is that I can deselect the Servlet API and the dependency goes away from Libraries. I'm guessing that was already the behaviour, right? We were just auto-selecting the API for new projects/add framework support.

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java]: Yea, you could previously unselect the servlet API from the facet editor to remove it.
